crates/experimentation_platform/src/api/experiments/handlers.rs

```diff
@@ -234,30 +244,48 @@ async fn create(
     // creating variants' context in CAC
     let http_client = reqwest::Client::new();
     let url = state.cac_host.clone() + "/context/bulk-operations";
-    let headers_map = construct_header_map(tenant.as_str(), custom_headers.config_tags)?;
+    let user_str = serde_json::to_string(&user).map_err(|err| {
```

Comment:
What are we trying to achieve with `x-user` header?

Reply:
Someone who has access for only making changes in experimentation and not in context-aware-config, for them all requests of experimenation would start failing as the when these api calls would be made to make changes in context-aware-config, the authorization would fail due to lack of access.

So Internal authorization has been added
Now since this Internal auth would not have the details of the current user, and on validating it would give the details of a super-user then the audit logs would show wrong info regarding the maker of the contexts in context-aware-config, so in this case since experimentation has already decoded the user, it just forwards the user data 

And this user data is only to be consumed if the Internal auth check is successful